yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -900,74 +889,40 @@ func (p *ParserIssueDetector) DetectMissingForeignKeyIndexes() []QueryIssue {
 // FK (x,y,z) → Index (x,y) ✗ (missing column) → Detected as missing index
 // FK (x,y,z) → No index ✗ → Detected as missing index
 func (p *ParserIssueDetector) hasProperIndexCoverage(fk ForeignKeyConstraint) bool {
-	// Use the existing detection logic from detectors_ddl.go
-	// We'll reuse the same functions but with the stored data
-
-	// Check for exact match first (O(1) lookup)
-	if p.hasExactMatch(fk.TableName, fk.ColumnNames) {
-		return true
-	}
-
-	// Check for permutation match (different column order)
-	if p.hasPermutationMatch(fk.TableName, fk.ColumnNames) {
-		return true
+	indexes, exists := p.tableIndexes[fk.TableName]
+	if !exists {
+		return false
 	}
 
-	// No suitable index found
-	return false
-}
-
-// hasExactMatch checks for exact column order match (O(1) lookup).
-func (p *ParserIssueDetector) hasExactMatch(tableName string, columns []string) bool {
-	key := fmt.Sprintf("%s.%s", tableName, strings.Join(columns, ","))
-	return p.indexPrefixCoverageMap[key]
-}
-
-// hasPermutationMatch checks for different column order (O(n!) for n columns).
-func (p *ParserIssueDetector) hasPermutationMatch(tableName string, columns []string) bool {
-	permutations := p.generatePermutations(columns)
-	for _, perm := range permutations {
-		key := fmt.Sprintf("%s.%s", tableName, strings.Join(perm, ","))
-		if p.indexPrefixCoverageMap[key] {
+	for _, index := range indexes {
+		if p.hasIndexCoverage(index, fk.ColumnNames) {
 			return true
 		}
 	}
+
 	return false
 }
 
-// generatePermutations generates all permutations of the given slice using a backtracking algorithm.
-// O(n!) complexity for n elements
-func (p *ParserIssueDetector) generatePermutations(elements []string) [][]string {
-	if len(elements) <= 1 {
-		return [][]string{elements}
-	}
-
-	var result [][]string
-	var backtrack func(used []bool, current []string)
-
-	backtrack = func(used []bool, current []string) {
-		if len(current) == len(elements) {
-			// Make a copy of current
-			perm := make([]string, len(current))
-			copy(perm, current)
-			result = append(result, perm)
-			return
+// hasIndexCoverage checks if an index provides coverage for the given foreign key columns.
+// It checks if the FK columns form a subset of the index columns (in any order) and are at the beginning.
+func (p *ParserIssueDetector) hasIndexCoverage(index *queryparser.Index, fkColumns []string) bool {
+	// Extract non-expression columns from the index
```

Comment:
what happens in case of expression columns? 
I am assuming we will not be considering that index. Is that in-line with what we want to do?